mysql/datadog_checks/mysql/mysql.py

```diff
@@ -235,102 +235,39 @@ def _collect_metrics(self, db, tags, options, queries, max_custom_queries):
         metrics.update(INNODB_VARS)
         metrics.update(BINLOG_VARS)
 
-        if is_affirmative(options.get('extra_status_metrics', False)):
+        if is_affirmative(self.config.options.get('extra_status_metrics', False)):
             self.log.debug("Collecting Extra Status Metrics")
             metrics.update(OPTIONAL_STATUS_VARS)
 
             if self.version.version_compatible((5, 6, 6)):
                 metrics.update(OPTIONAL_STATUS_VARS_5_6_6)
 
-        if is_affirmative(options.get('galera_cluster', False)):
+        if is_affirmative(self.config.options.get('galera_cluster', False)):
             # already in result-set after 'SHOW STATUS' just add vars to collect
             self.log.debug("Collecting Galera Metrics.")
             metrics.update(GALERA_VARS)
 
         performance_schema_enabled = self._get_variable_enabled(results, 'performance_schema')
         above_560 = self.version.version_compatible((5, 6, 0))
-        if is_affirmative(options.get('extra_performance_metrics', False)) and above_560 and performance_schema_enabled:
+        if (
+            is_affirmative(self.config.options.get('extra_performance_metrics', False))
+            and above_560
+            and performance_schema_enabled
+        ):
             # report avg query response time per schema to Datadog
             results['perf_digest_95th_percentile_avg_us'] = self._get_query_exec_time_95th_us(db)
             results['query_run_time_avg'] = self._query_exec_time_per_schema(db)
             metrics.update(PERFORMANCE_VARS)
 
-        if is_affirmative(options.get('schema_size_metrics', False)):
+        if is_affirmative(self.config.options.get('schema_size_metrics', False)):
             # report avg query response time per schema to Datadog
             results['information_schema_size'] = self._query_size_per_schema(db)
             metrics.update(SCHEMA_VARS)
 
-        if is_affirmative(options.get('replication', False)):
-            # Get replica stats
-            is_mariadb = self.version.flavor == "MariaDB"
-            replication_channel = options.get('replication_channel')
-            if replication_channel:
-                self.service_check_tags.append("channel:{0}".format(replication_channel))
```

Comment:
I don't think this operation was ported in this new version. I assume that's because we do `tags + service_check_tags` when submitting service checks, so `channel` is still submitted.

Reply:
Exactly, now `config.tags` has channel [from the beginning](https://github.com/DataDog/integrations-core/pull/8380/files#diff-00c1b706e1d0a42a871951aaaed96f51970dd636281c442b67f21e6fbedb34b2R25) so `service_check_tags` gets it [when built](https://github.com/DataDog/integrations-core/pull/8380/files#diff-4ecc52a59650f90cba9b1b1b3a145f8dc3457d133fc52419846c63e62bbe9bd8R177).

Reply:
No, seems the only place was https://github.com/DataDog/integrations-core/pull/8380/files#diff-4ecc52a59650f90cba9b1b1b3a145f8dc3457d133fc52419846c63e62bbe9bd8R389 and we are ensuring tags are unique there